libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpMPEG4Reader.java

```diff
@@ -109,20 +116,23 @@ public void seek(long nextRtpTimestamp, long timeUs) {
   // Internal methods.
 
   /**
-   * Parses VOP Coding type
+   * Parses VOP Coding type.
    *
    * Sets {@link #bufferFlags} according to the VOP Coding type.
    */
-  private void parseVopType(ParsableByteArray data) {
+  @C.BufferFlags
+  private static int getBufferFlagsFromVop(ParsableByteArray data) {
+    int flags = 0;
     // search for VOP_START_CODE (00 00 01 B6)
     byte[] inputData = data.getData();
-    byte[] startCode = {0x0, 0x0, 0x01, (byte) 0xB6};
+    byte[] startCode = new byte[] {0x0, 0x0, 0x1, (byte) 0xB6};
     int vopStartCodePos = Bytes.indexOf(inputData, startCode);
     if (vopStartCodePos != -1) {
       data.setPosition(vopStartCodePos + 4);
       int vopType = data.peekUnsignedByte() >> 6;
-      bufferFlags = getBufferFlagsFromVopType(vopType);
+      flags = vopType == I_VOP ? C.BUFFER_FLAG_KEY_FRAME : 0;
     }
+    return flags;
```

Comment:
I think we can remove the `flags` and return `vopType == I_VOP ? C.BUFFER_FLAG_KEY_FRAME : 0` in the if, and 0 on the outside